Optimized DSO textures

sunshuwei opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

Remove star points from large deep sky object textures to achieve better visual effects.
Optimize the visual effects of M4 and M80.
Losslessly compressed galaxy textures.

![m4](https://github.com/user-attachments/assets/cd2d8009-1824-4a68-9e5a-420d76ea5608)
![m80](https://github.com/user-attachments/assets/9e217ed6-033d-439d-aaf7-41340dbd349f)
![0ca9c489-309c-403b-8328-71745ce7b678](https://github.com/user-attachments/assets/72507cfe-0949-479f-8429-f3e7fb342f4c)
![9ec02cb4-ae0e-4852-8b44-a97d98569648](https://github.com/user-attachments/assets/39ed74a3-f2b4-4bcf-93e7-e30852ef9234)
![28382c3b-647e-45be-a725-fb407d4d1d1e](https://github.com/user-attachments/assets/b00ce45f-ccd1-4f52-8cb3-87b1554113e8)
![c189e00a-328c-4163-a4b2-b114fbc3b87b](https://github.com/user-attachments/assets/2baf5f9c-8c09-4934-8ce1-6a6a189854c7)



### Description
<!--- Please include a summary of the change and which issue is fixed. -->
<!--- Please also include relevant motivation and context. -->
<!--- List any dependencies that are required for this change. -->

Fixes # (issue)

### Screenshots (if appropriate):

### Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [ ] Housekeeping

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

**Test Configuration**:
* Operating system: <Name, version number>
* Graphics Card: <Manufacturer (likely Intel, NVidia, AMD?), Model (HD, Geforce, Radeon..., with model number), driver version?>

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

10110111 commented:
Does it actually make sense to remove stars surrounding star clusters like M4? Also, you seem to have removed some background glow from it — is it not part of the object? It seems to also be present in the [eso1235a](https://www.eso.org/public/images/eso1235a/) shot of it.

10110111 commented:
> Also, you seem to have removed some background glow from it — is it not part of the object?

Oh, I get it: the cluster resides inside a nebula that's present in a separate texture (`rho-a*.png`).

sunshuwei commented:
I have restored the Rho image because removing the star point will mistakenly delete NGC6144.
So I restored M4 and M80 as well.

There are two images at the position of M4. Therefore, it is too bright. If you remove M4.png, the visual effect is like this.

![微信图片_20241207135313](https://github.com/user-attachments/assets/6acdea5e-c131-4373-88ed-5eee171c0083)
If you want to keep M4.PNG and remove M4 on the RHO image, it will be like this.
![微信图片_20241207142237](https://github.com/user-attachments/assets/30549762-73b7-4e1f-9083-15583a543dca)
If you keep both, it will be as shown below, with a strange visual effect.
![微信图片_20241207144154](https://github.com/user-attachments/assets/ac014fbd-c454-443c-b935-4847a174b70a)

M80 also has two images and appears abnormally bright.

sunshuwei commented:
> > Also, you seem to have removed some background glow from it — is it not part of the object?
> 
> Oh, I get it: the cluster resides inside a nebula that's present in a separate texture (`rho-a*.png`).

Yes,
I deleted the part about RHO, M4, and M80. We can solve this problem separately with a new PR.

10110111 commented:
The change to sh2-264 looks too crude. Two circles are visible at the location of the two removed nebulas, the one around Bellatrix is especially obvious.

gzotti commented:
In principle I welcome the removal of stars in the DSO images once they have been fit in. The globulars may pose a problem in that an AI "star remover" may not catch all the right non-cluster stars, but a circular fading mask will likely do to mitigate the "image rectangle" effect. A remaining "ring" of catalog stars in its outer (still resolved) regions (M80 example) always looks bad, though.

sunshuwei commented:
> In principle I welcome the removal of stars in the DSO images once they have been fit in. The globulars may pose a problem in that an AI "star remover" may not catch all the right non-cluster stars, but a circular fading mask will likely do to mitigate the "image rectangle" effect. A remaining "ring" of catalog stars in its outer (still resolved) regions (M80 example) always looks bad, though.

The "ring" is not caused by the image, but by the star catalog. Many globular clusters have this ring. Hopefully the new catalog will solve this problem。
![df69ce80-4691-4251-8647-d45dec61c414](https://github.com/user-attachments/assets/c2f8d9ba-683e-4eba-9017-dd71a58bcb24)

sunshuwei commented:
I have uploaded the texture of NGC6144.And used the new M4 textures. Now it looks like this.
![Uploading 41afccec-9194-4bca-ba18-c0ff1d37a0a1.png…]()

gzotti commented:
I know. However, in default star brightness settings, the cluster in the image should fill the inside of the ring, making it invisible.

sunshuwei commented:
M80 looks like this now.
![dc3ddde5-78c9-4aa0-b2b4-58dd3cc2fd3f](https://github.com/user-attachments/assets/5f1ed931-4a19-41ce-be9d-bcf854298bb9)

sunshuwei commented:
Added textures for the Dark Doodad Nebula
The previous texture of NGC 4372 is a DSS image. It has a strong interaction with the Dark Doodad Nebula and should appear together. They are very close in the night sky, and the dust ofthe Dark Doodad Nebula has some obstruction on  NGC 4372.

![a7a9ffbd-a403-4ba4-865d-65c265aa29e7](https://github.com/user-attachments/assets/8d5a8be7-12e6-400a-ab63-f4296aae419a)

sunshuwei commented:
The texture of NGC1365 also overlaps.
Cut off the low-quality overlapping NGC 1365 texture on fornax_cluster.png. 
Use the new and better NGC1365 texture. 
Delete the low-quality NGC1365 texture.
It looks good now, without any overexposure issues.
![微信图片_20241213114607](https://github.com/user-attachments/assets/fb4f8e2d-a386-4639-b59e-d09e62ec78ec)

alex-w commented:
I fear we should target this PR to milestone 25.1

sunshuwei commented:
This is the new IC2177 texture.


![ic2177 示意](https://github.com/user-attachments/assets/1aff97af-bd46-43b2-b216-9f746d0eb87f)
![ic2177 示意图](https://github.com/user-attachments/assets/9a4a0fce-36fa-42e2-bc35-b299fd29091f)

The following is how it looks without removing the star points. I made strict alignment and then removed the star points.
![ic2177 没有去处恒星](https://github.com/user-attachments/assets/88596802-e1bb-40ee-93cd-0c2eda67c85c)

![ic2177去星](https://github.com/user-attachments/assets/1cb846e7-a82e-4b92-8aa8-50864d8e606d)

sunshuwei commented:
> I fear we should target this PR to milestone 25.1
You can merge it at any time you like, and you can test it to see the effect. It does not conflict with other branches.

sunshuwei commented:
> The change to sh2-264 looks too crude. Two circles are visible at the location of the two removed objects, the one around Bellatrix is especially obvious.

I have reprocessed SH2-264.

sunshuwei commented:
> In principle I welcome the removal of stars in the DSO images once they have been fit in. The globulars may pose a problem in that an AI "star remover" may not catch all the right non-cluster stars, but a circular fading mask will likely do to mitigate the "image rectangle" effect. A remaining "ring" of catalog stars in its outer (still resolved) regions (M80 example) always looks bad, though.

Some images with too large angular diameters have too large star points, which not only look dirty after zooming in, but also cover "small-scale" deep sky objects, such as planetary nebulae. And now the new Gaia star catalog should have enough stars.

alex-w commented:
@sunshuwei is the PR ready to review?

sunshuwei commented:
> @sunshuwei is the PR ready to review?

Yes, please review.

sunshuwei commented:
Some globular cluster textures, mainly from the southern sky, come from my friend Tian Mai. These targets are relatively remote and have fewer photos.
![NGC6397](https://github.com/user-attachments/assets/102b1917-ee5b-4b14-82cb-c4679df91a21)
![NGC6541](https://github.com/user-attachments/assets/8dfed3f0-6aeb-44ec-af01-b94fecd3aca4)
![NGC6752](https://github.com/user-attachments/assets/83ed3fec-2650-444b-a476-019d9ff4448b)
![NGC1261](https://github.com/user-attachments/assets/de8b8706-b607-4213-9d8b-5a5792da1cd3)
![NGC3201](https://github.com/user-attachments/assets/1736bc7a-b2d4-4e6f-b4e4-f08266ef9f3a)
![NGC5286](https://github.com/user-attachments/assets/861e8f94-8dd7-4816-9a95-a6d1de4d82bf)
![NGC6101](https://github.com/user-attachments/assets/bd177ff9-54f0-4bc6-a997-d073c71e489a)

sunshuwei commented:
This is a new C68 texture, and the globular cluster NGC6723 on the texture has a good appearance.
![微信图片_20241226000518](https://github.com/user-attachments/assets/12defb70-e210-4e6e-a457-0a3287189dfa)
![微信图片_20241226000554](https://github.com/user-attachments/assets/e1981404-c0f8-4fc4-a0b2-219769161bd2)

alex-w reviewed (changes requested):
NGC 6934, M49, M59, M60 - no info in `CREDITS` file; 

UGC 10822 - typo in `CREDITS` file; 

sh2-313, sh2-129 - typo in `textures.json` file or in name of texture (case sensitive file names!)

sunshuwei commented:
> NGC 6934, M49, M59, M60 - no info in `CREDITS` file;
> 
Fixed.
> UGC 10822 - typo in `CREDITS` file;
> 
Fixed.
> sh2-313, sh2-129 - typo in `textures.json` file or in name of texture (case sensitive file names!)
> 
Fixed.
